samples/snippets/csharp/programming-guide/classes-and-structs/object-collection-initializers/HowToClassTypedInitializer.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace object_collection_initializers
+{
+    // <SnippetHowToClassTypedInitializer>
+    public class HowToClassTypedInitializer
+    {
+        public class EmbeddedClassTypeA
+        {
+            public int I { get; set; }
+            public bool B { get; set; }
+            public string S { get; set; }
+            public EmbeddedClassTypeB ClassB { get; set; }
+
+            public override string ToString() => $"{I}|{B}|{S}|||{ClassB}";
+
+            public EmbeddedClassTypeA()
+            {
+                I = 3;
+                B = true;
+                S = "abc";
+                ClassB = new() { BB = true, BI = 43 };
+            }
+        }
+
+        public class EmbeddedClassTypeB
+        {
+            public int BI { get; set; }
+            public bool BB { get; set; }
+            public string BS { get; set; }
+
+            public override string ToString() => $"{BI}|{BB}|{BS}";
+
+            public EmbeddedClassTypeB()
+            {
+                BI = 23;
+                BB = false;
+                BS = "BBBabc";
+            }
```

Comment:
Here's one of the locations where I'd add information about the properties to show the order of initialization:

```suggestion
            public EmbeddedClassTypeB()
            {
               Console.WriteLine($"Entering EmbeddedClassTypeB constructor. Values are: {this}");
                BI = 23;
                BB = false;
                BS = "BBBabc";
                Console.WriteLine($"Exiting EmbeddedClassTypeB constructor. Values are: {this})");
            }
```

Then, continue that process with the other constructors, and in the `Main` method after the initialization happens.